app/lib/metrics_service.rb

```diff
@@ -90,28 +93,24 @@ def self.merge_identities_in_mixpanel(user, cookies={})
   # @param {String} name Name of the event
   # @param {Hash} props Properties associated with the event
   # @param {User} user User model object
-  def self.log(name, props={}, user)
+  def self.log(name, props={}, user=nil, request: nil)
     Rails.logger.info "#{Time.zone.now}: Logging analytics to Mixpanel for event name: #{name}"
 
     props.merge!({
       appId: 'single-cell-portal',
       env: Rails.env,
-      registeredForTerra: user.registered_for_firecloud,
       logger: 'app-backend'
     })
 
     headers = get_default_headers(user)
 
-    access_token = user.token_for_api_call
-    user_id = user.get_metrics_uuid
-
-    if access_token.nil? || !user.registered_for_firecloud
-      # User is unauthenticated / unregistered / anonymous
-      props['distinct_id'] = user_id
-      headers.delete('Authorization')
-      props['authenticated'] = false
+    # configure properties/headers depending on user presence
+    # only pass user token if user is registered for Terra to avoid 4xx/5xx errors
+    if user.present? && user.registered_for_firecloud
+      props.merge!({ authenticated: true, registeredForTerra: user.registered_for_firecloud })
     else
-      props['authenticated'] = true
+      distinct_id = user&.get_metrics_uuid || request&.cookies['user_id']
+      props.merge!({ authenticated: false, distinct_id: distinct_id, registeredForTerra: false })
```

Comment:
More succinctly, and easier to compare to the similar command on line 110:

```suggestion
      props.merge!({ authenticated: false, registeredForTerra: false, distinct_id: })
```